File: bindings/kotlin/src/main/kotlin/io/ed3c/aiedge/contracts/v1/Enums.kt
package io.ed3c.aiedge.contracts.v1

enum class ApiMaturity(val wireValue: String) {
    STABLE("STABLE"),
    PREVIEW("PREVIEW"),
    BETA("BETA"),
    COMMUNITY("COMMUNITY"),
    DISABLED("DISABLED");

    companion object { fun fromWire(value: String): ApiMaturity = entries.first { it.wireValue == value } }
}

enum class ArtifactFormat(val wireValue: String) {
    LITERTLM("LITERTLM"),
    TASK("TASK"),
    COREML_PACKAGE("COREML_PACKAGE"),
    ONNX("ONNX"),
    OTHER("OTHER");

    companion object { fun fromWire(value: String): ArtifactFormat = entries.first { it.wireValue == value } }
}

enum class BackendKind(val wireValue: String) {
    CPU("CPU"),
    GPU("GPU"),
    NPU("NPU"),
    ANE("ANE"),
    NNAPI("NNAPI"),
    METAL("METAL"),
    WEBGPU("WEBGPU"),
    UNKNOWN("UNKNOWN");

    companion object { fun fromWire(value: String): BackendKind = entries.first { it.wireValue == value } }
}

enum class ErrorCode(val wireValue: String) {
    CAPABILITY_UNAVAILABLE("CAPABILITY_UNAVAILABLE"),
    POLICY_DENIED("POLICY_DENIED"),
    ARTIFACT_INVALID("ARTIFACT_INVALID"),
    MODEL_LOAD_FAILED("MODEL_LOAD_FAILED"),
    GENERATION_FAILED("GENERATION_FAILED"),
    SCHEMA_INVALID("SCHEMA_INVALID"),
    TOOL_DENIED("TOOL_DENIED"),
    TOOL_FAILED("TOOL_FAILED"),
    VALIDATION_FAILED("VALIDATION_FAILED"),
    TIMEOUT("TIMEOUT"),
    RESOURCE_EXHAUSTED("RESOURCE_EXHAUSTED"),
    CANCELLED("CANCELLED"),
    EXHAUSTED_FALLBACK("EXHAUSTED_FALLBACK"),
    INTERNAL("INTERNAL");

    companion object { fun fromWire(value: String): ErrorCode = entries.first { it.wireValue == value } }
}

enum class EvidenceLane(val wireValue: String) {
    SOURCE("SOURCE"),
    STATIC("STATIC"),
    LOCAL("LOCAL"),
    LIVE_DEVICE("LIVE_DEVICE"),
    PRIVATE("PRIVATE"),
    HUMAN("HUMAN");

    companion object { fun fromWire(value: String): EvidenceLane = entries.first { it.wireValue == value } }
}

enum class EvidenceState(val wireValue: String) {
    PASS("PASS"),
    FAIL("FAIL"),
    ABSENT("ABSENT"),
    NOT_IMPLEMENTED("NOT_IMPLEMENTED"),
    NOT_EXERCISED("NOT_EXERCISED"),
    SKIPPED_BY_POLICY("SKIPPED_BY_POLICY"),
    HUMAN_ADMIT_REQUIRED("HUMAN_ADMIT_REQUIRED");

    companion object { fun fromWire(value: String): EvidenceState = entries.first { it.wireValue == value } }
}

enum class InferenceEventType(val wireValue: String) {
    STARTED("STARTED"),
    TOKEN_DELTA("TOKEN_DELTA"),
    TOOL_PROPOSAL("TOOL_PROPOSAL"),
    TOOL_RESULT("TOOL_RESULT"),
    COMPLETED("COMPLETED"),
    DEGRADED("DEGRADED"),
    FAILED("FAILED"),
    CANCELLED("CANCELLED");

    companion object { fun fromWire(value: String): InferenceEventType = entries.first { it.wireValue == value } }
}

enum class LicensePlane(val wireValue: String) {
    SOURCE_CODE("SOURCE_CODE"),
    MODEL_WEIGHTS("MODEL_WEIGHTS"),
    DATASET("DATASET"),
    SERVICE("SERVICE"),
    SDK_STORE("SDK_STORE"),
    TRADEMARK("TRADEMARK"),
    EXPORT_CONTROL("EXPORT_CONTROL"),
    UNKNOWN("UNKNOWN");

    companion object { fun fromWire(value: String): LicensePlane = entries.first { it.wireValue == value } }
}

enum class MessageRole(val wireValue: String) {
    SYSTEM("SYSTEM"),
    USER("USER"),
    ASSISTANT("ASSISTANT"),
    TOOL("TOOL");

    companion object { fun fromWire(value: String): MessageRole = entries.first { it.wireValue == value } }
}

enum class Modality(val wireValue: String) {
    TEXT("TEXT"),
    IMAGE("IMAGE"),
    AUDIO("AUDIO");

    companion object { fun fromWire(value: String): Modality = entries.first { it.wireValue == value } }
}

enum class PlatformKind(val wireValue: String) {
    ANDROID("ANDROID"),
    IOS("IOS"),
    MACOS("MACOS"),
    LINUX("LINUX"),
    WINDOWS("WINDOWS"),
    WEB("WEB"),
    EMBEDDED("EMBEDDED");

    companion object { fun fromWire(value: String): PlatformKind = entries.first { it.wireValue == value } }
}

enum class ProviderKind(val wireValue: String) {
    SYSTEM_MODEL("SYSTEM_MODEL"),
    EMBEDDED_MODEL("EMBEDDED_MODEL"),
    CLOUD("CLOUD");

    companion object { fun fromWire(value: String): ProviderKind = entries.first { it.wireValue == value } }
}

enum class ResultState(val wireValue: String) {
    SUCCEEDED("SUCCEEDED"),
    FAILED("FAILED"),
    CANCELLED("CANCELLED");

    companion object { fun fromWire(value: String): ResultState = entries.first { it.wireValue == value } }
}

enum class SkillTrustState(val wireValue: String) {
    TRUSTED("TRUSTED"),
    UNTRUSTED("UNTRUSTED"),
    REJECTED("REJECTED");

    companion object { fun fromWire(value: String): SkillTrustState = entries.first { it.wireValue == value } }
}

enum class TaskKind(val wireValue: String) {
    CHAT("CHAT"),
    SUMMARIZE("SUMMARIZE"),
    STRUCTURED_GENERATION("STRUCTURED_GENERATION"),
    FUNCTION_CALLING("FUNCTION_CALLING"),
    VISION("VISION"),
    ASR("ASR"),
    TEXT_POLISHING("TEXT_POLISHING"),
    EMBEDDING("EMBEDDING"),
    CUSTOM("CUSTOM");

    companion object { fun fromWire(value: String): TaskKind = entries.first { it.wireValue == value } }
}

enum class TermsState(val wireValue: String) {
    NOT_REQUIRED("NOT_REQUIRED"),
    REVIEW_REQUIRED("REVIEW_REQUIRED"),
    HUMAN_ADMIT_REQUIRED("HUMAN_ADMIT_REQUIRED"),
    ACCEPTED("ACCEPTED"),
    REJECTED("REJECTED");

    companion object { fun fromWire(value: String): TermsState = entries.first { it.wireValue == value } }
}

enum class ToolDecision(val wireValue: String) {
    ALLOW("ALLOW"),
    DENY("DENY"),
    REQUIRE_CONFIRMATION("REQUIRE_CONFIRMATION");

    companion object { fun fromWire(value: String): ToolDecision = entries.first { it.wireValue == value } }
}

enum class ToolEffect(val wireValue: String) {
    PURE("PURE"),
    READ_LOCAL("READ_LOCAL"),
    WRITE_LOCAL("WRITE_LOCAL"),
    EXTERNAL_SIDE_EFFECT("EXTERNAL_SIDE_EFFECT");

    companion object { fun fromWire(value: String): ToolEffect = entries.first { it.wireValue == value } }
}

File: bindings/swift/Sources/AIEdgeContracts/Enums.swift
import Foundation

public enum ApiMaturity: String, Codable, Equatable, Sendable {
    case stable = "STABLE"
    case preview = "PREVIEW"
    case beta = "BETA"
    case community = "COMMUNITY"
    case disabled = "DISABLED"
}

public enum ArtifactFormat: String, Codable, Equatable, Sendable {
    case litertlm = "LITERTLM"
    case task = "TASK"
    case coremlPackage = "COREML_PACKAGE"
    case onnx = "ONNX"
    case other = "OTHER"
}

public enum BackendKind: String, Codable, Equatable, Sendable {
    case cpu = "CPU"
    case gpu = "GPU"
    case npu = "NPU"
    case ane = "ANE"
    case nnapi = "NNAPI"
    case metal = "METAL"
    case webgpu = "WEBGPU"
    case unknown = "UNKNOWN"
}

public enum ErrorCode: String, Codable, Equatable, Sendable {
    case capabilityUnavailable = "CAPABILITY_UNAVAILABLE"
    case policyDenied = "POLICY_DENIED"
    case artifactInvalid = "ARTIFACT_INVALID"
    case modelLoadFailed = "MODEL_LOAD_FAILED"
    case generationFailed = "GENERATION_FAILED"
    case schemaInvalid = "SCHEMA_INVALID"
    case toolDenied = "TOOL_DENIED"
    case toolFailed = "TOOL_FAILED"
    case validationFailed = "VALIDATION_FAILED"
    case timeout = "TIMEOUT"
    case resourceExhausted = "RESOURCE_EXHAUSTED"
    case cancelled = "CANCELLED"
    case exhaustedFallback = "EXHAUSTED_FALLBACK"
    case _internal = "INTERNAL"
}

public enum EvidenceLane: String, Codable, Equatable, Sendable {
    case source = "SOURCE"
    case _static = "STATIC"
    case local = "LOCAL"
    case liveDevice = "LIVE_DEVICE"
    case _private = "PRIVATE"
    case human = "HUMAN"
}

public enum EvidenceState: String, Codable, Equatable, Sendable {
    case pass = "PASS"
    case fail = "FAIL"
    case absent = "ABSENT"
    case notImplemented = "NOT_IMPLEMENTED"
    case notExercised = "NOT_EXERCISED"
    case skippedByPolicy = "SKIPPED_BY_POLICY"
    case humanAdmitRequired = "HUMAN_ADMIT_REQUIRED"
}

public enum InferenceEventType: String, Codable, Equatable, Sendable {
    case started = "STARTED"
    case tokenDelta = "TOKEN_DELTA"
    case toolProposal = "TOOL_PROPOSAL"
    case toolResult = "TOOL_RESULT"
    case completed = "COMPLETED"
    case degraded = "DEGRADED"
    case failed = "FAILED"
    case cancelled = "CANCELLED"
}

public enum LicensePlane: String, Codable, Equatable, Sendable {
    case sourceCode = "SOURCE_CODE"
    case modelWeights = "MODEL_WEIGHTS"
    case dataset = "DATASET"
    case service = "SERVICE"
    case sdkStore = "SDK_STORE"
    case trademark = "TRADEMARK"
    case exportControl = "EXPORT_CONTROL"
    case unknown = "UNKNOWN"
}

public enum MessageRole: String, Codable, Equatable, Sendable {
    case system = "SYSTEM"
    case user = "USER"
    case assistant = "ASSISTANT"
    case tool = "TOOL"
}

public enum Modality: String, Codable, Equatable, Sendable {
    case text = "TEXT"
    case image = "IMAGE"
    case audio = "AUDIO"
}

public enum PlatformKind: String, Codable, Equatable, Sendable {
    case android = "ANDROID"
    case ios = "IOS"
    case macos = "MACOS"
    case linux = "LINUX"
    case windows = "WINDOWS"
    case web = "WEB"
    case embedded = "EMBEDDED"
}

public enum ProviderKind: String, Codable, Equatable, Sendable {
    case systemModel = "SYSTEM_MODEL"
    case embeddedModel = "EMBEDDED_MODEL"
    case cloud = "CLOUD"
}

public enum ResultState: String, Codable, Equatable, Sendable {
    case succeeded = "SUCCEEDED"
    case failed = "FAILED"
    case cancelled = "CANCELLED"
}

public enum SkillTrustState: String, Codable, Equatable, Sendable {
    case trusted = "TRUSTED"
    case untrusted = "UNTRUSTED"
    case rejected = "REJECTED"
}

public enum TaskKind: String, Codable, Equatable, Sendable {
    case chat = "CHAT"
    case summarize = "SUMMARIZE"
    case structuredGeneration = "STRUCTURED_GENERATION"
    case functionCalling = "FUNCTION_CALLING"
    case vision = "VISION"
    case asr = "ASR"
    case textPolishing = "TEXT_POLISHING"
    case embedding = "EMBEDDING"
    case custom = "CUSTOM"
}

public enum TermsState: String, Codable, Equatable, Sendable {
    case notRequired = "NOT_REQUIRED"
    case reviewRequired = "REVIEW_REQUIRED"
    case humanAdmitRequired = "HUMAN_ADMIT_REQUIRED"
    case accepted = "ACCEPTED"
    case rejected = "REJECTED"
}

public enum ToolDecision: String, Codable, Equatable, Sendable {
    case allow = "ALLOW"
    case deny = "DENY"
    case requireConfirmation = "REQUIRE_CONFIRMATION"
}

public enum ToolEffect: String, Codable, Equatable, Sendable {
    case pure = "PURE"
    case readLocal = "READ_LOCAL"
    case writeLocal = "WRITE_LOCAL"
    case externalSideEffect = "EXTERNAL_SIDE_EFFECT"
}
